lib/zip/file.rb

```diff
@@ -406,7 +405,7 @@ def on_success_replace
       tmp_filename = create_tmpname
       if yield tmp_filename
         ::File.rename(tmp_filename, name)
-        ::File.chmod(@file_permissions, name) if defined?(@file_permissions)
+        ::File.chmod(@file_permissions, name) if @create.nil?
```

Comment:
Why change this to `if @create.nil?` instead of `defined?(@file_permissions)`

With your removal of the `@file_permissions` setter in the `when create` branch should mean that that variable is only-ever set for the non-create path. 

I also lean towards avoiding a `.nil?` check on the `@create` variable, because the desired values for `@create` are a bit fuzzy to me, as a developer who's a bit unfamiliar with the code. The code comment for the `initialize` method suggests that `@create` should be either `nil` or `true`, but the constant is `CREATE = 1` 😃